atomate/atat/fireworks/core.py

```diff
@@ -0,0 +1,193 @@
+# coding: utf-8
+
+from __future__ import absolute_import, division, print_function, \
+    unicode_literals
+
+try:
+    from math import gcd
+except ImportError:
+    from fractions import gcd
+
+import multiprocessing
+import os
+import json
+from fireworks import Firework
+from fireworks import ScriptTask, FileWriteTask
+from monty.os.path import which
+from atomate.utils.utils import get_logger
+from atomate.atat.firetasks.parse_outputs import McsqsToDbTask
+from atomate.atat.utilities import preprocess_disordered_structure
+from fractions import Fraction
+from string import Template
+
+__author__ = 'Matthew Horton'
+__credits__ = 'Josua Vieten'
+__email__ = 'mkhorton@lbl.gov'
+
+logger = get_logger(__name__)
+module_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)))
+
+class McsqsFW(Firework):
+
+    def __init__(self, disordered_struct,
+                 name="mcsqs",
+                 size=None,
+                 clusters=None,
+                 user_input_settings=None,
+                 max_denominator=8,
+                 walltime=60,
+                 ncores=None,
+                 db_file=None,
+                 **kwargs):
+        """
+        Find a best SQS approximation for a given disordered
+        structure using the mcsqs code from the ATAT library.
+
+        One unusual feature of this Firework: walltime must
+        be specified. This is because there are often cases
+        where mcsqs will never terminate (it will keep trying
+        structures at random); however, whatever structure
+        it finds in this time (the 'bestsqs') is often
+        still useful.
+
+        :param disordered_struct: disordered structure
+        :param name: name of the Firework
+        :param size: ratio of atoms in the output supercell
+        to number of atoms in input cell, if None will
+        choose the smallest possible supercell
+        :param clusters: dict of cluster sizes, e.g.
```

Comment:
I am a bit afraid that the clusters parameters (if set manually and rather than using the default of None) will not serialize and deserialize properly. In particular, Mongo doesn't like int keys for dicts and we end up having to store such keys as strings and convert back to int() in the from_dict(). Would check this ...

Reply:
(this is limitation is in the JSON spec?)